### DISCUSSION TITLE
[Slugger] Adding a characterMap

### DISCUSSION THREAD

**Discussion by florianhuber81** [symfony/symfony#46169] [Q&A]
Thanks for having invented the slugger. A great feature!

How about adding some kind of characterMap for certain characters to be ignored when sanitizing the given string. For filenames I often need underscores and dashes. IMO that is not possible at the moment. All non [^A-Za-z0-9] characters get converted to the defined separator. With a characterMap there could be some exceptions.

eg.: monthly_report-2022-04-25.pdf should be ok but is converted to monthly_report_2022_04_25.pdf 

This also makes it harder to for example explode() these filenames for whatever reason.

**Comment by OskarStark** [comment]
@nicolas-grekas as you provided the symfony/string component, any thoughts on this topic?

**Comment by nicolas-grekas** [comment]
We could make the regexp configurable via a constructor argument yes.
